### DISCUSSION TITLE
BriefcaseManager.downloadBriefcase error 90113

### DISCUSSION THREAD

**Discussion by manderson88** [iTwin/itwinjs-core#4782] [Q&A]
I have a process that downloads a briefcase and processes the components in it.  We have this working successfully on a number of projects.  Today (12/8) one of the imodel projects started failing.  Upon investigation I am seeing that the download starts but then fails after 120 changesets with an error message 90113 'error applying changeset at function applySingleChangeset... ' Where can I find some information on this error? And more important what can I do to stop it from happening?  I am using the v2.x generation api (long story).

thanks,
marka

**Comment by nick4598** [comment]
It sounds like there may be something wrong with that particular changeset. Could you add a few lines of code to your process which is downloading the briefcase and applying changesets and then rerun that process?

Logger.initializeToConsole(); (If you already have your Logger set up properly, this line is not necessary).
Logger.setLevel(NativeLoggerCategory.DgnCore, LogLevel.Warning);

The Logger class and LogLevel enum  comes from "@bentley/bentleyjs-core" and the NativeLoggerCategory enum comes from "@bentley/imodeljs-backend".

This should hopefully output some more information pertaining to what went wrong when applying that changeset.

**Reply by ColinKerr** [reply]
I suggest turning on the additional following categories:
```
NativeLoggerCategory.BeSQLite
NativeLoggerCategory.Changeset
NativeLoggerCategory.ECDb
NativeLoggerCategory.SQLite
```

This may give you a bit more than you need but it should ensure we get all the necessary logs.

**Reply by ColinKerr** [reply]
With the help of the SQLite team have identified this as an issue with SQLite ChangeSet apply code.  The SQLite team has pushed a [fix](https://sqlite.org/src/info/f9cd23dffba06b19) and we have merged the fix into our [C++ codebase](https://github.com/iTwin/imodel-native/pull/76#event-8050321645) and will push out a patch to iTwin.js 3.5.x as soon as we complete testing the changes.

We will update this discussion with the iTwin.js version once the packages are released.

**Reply by ColinKerr** [reply]
iTwin.js versions 3.5.2 and 2.19.49 include the updated version of SQLite with the fix.  With these versions the changesets should now apply correctly.


For a detailed description of the underlying bug checkout the related SQLite Change: https://sqlite.org/src/info/f9cd23dffba06b19

**Comment by fever-Wong** [comment]
谢谢，您发给我的邮件已经收到，我会尽快处理。Thank you,the email you sent me has been received and I will handle it as soon as possible.王景博fever wong
